extensions/service-binding.md

```diff
@@ -0,0 +1,66 @@
+# Service Binding
+
+## Buildpack Exposure
+
+During the detection and build phases, the lifecycle MUST provide any bound services as files in `<platform>/services/<service-name>` with directory names matching the name of the bound service.
```

Comment:
Do we need/want to have restrictions on the directory names here?

Reply:
Unclear to me.  Have we been putting limitations like this on other parts of the spec @sclevine?

Reply:
@nebhale is this the thing you're thinking about from the spec? https://github.com/buildpack/spec/blob/master/buildpack.md#buildpacktoml-toml

Reply:
@hone Yeah, something like that and there must be some bit that specifies how those directories are sanitized (the `/` specifically)?

Reply:
I'm thinking `[a-z0-9\-\.]{1,253}` for service names.  This matches the Kubernetes [object name restriction](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names) (which seems a reasonable start point).

Reply:
Most Helm charts truncate names to 63 chars, would that be too short? https://github.com/helm/charts/blob/master/stable/mysql/templates/_helpers.tpl#L11

Reply:
Ah, because of the DNS naming restriction.  I'd be open to going that small, but I'm not sure it's necessary.  Since it's a subset of the broader naming restriction, all those DNS-named services would be valid, and a larger size allows services that aren't DNS-limited would also be available.

But realistically, almost no one would ever touch 63 characters, either way.

Anyone have any opinions?